File: jorgecunha07-frontend-1dcc492a1c33/jorgecunha07-frontend-1dcc492a1c33/src/app/Mappers/SurveillanceTaskMapper.ts
import { TaskStatusEnum } from "../Constants/TastStatusEnum";
import SurveillanceTask_Model from "../models/SurveillanceTask_Model";
import SurveillanceTaskDTO from "../dto/SurveillanceTaskDTO";

// Assuming LocationDto and SurveillanceTaskDTO, SurveillanceTask_Model are imported or defined here

export class SurveillanceTaskMapper {
  public static toSurveillanceTaskDTO(model: SurveillanceTask_Model): SurveillanceTaskDTO {
    return {
      TaskState: model.TaskState,
      Description: model.Description,
      FromLocation: {
        Building: model.FromLocation.Building,
        Room: model.FromLocation.Room,
        X: model.FromLocation.X,
        Y: model.FromLocation.Y
      },
      ToLocation: { ...model.ToLocation },
      ContactInfo: model.ContactInfo,
      User: model.User,
      RobotId: model.RobotId,
      RobotType: model.RobotType,
      name: model.name
    };
  }

  public static toSurveillanceTaskModel(dto: SurveillanceTaskDTO): SurveillanceTask_Model {
    return {
      TaskState: dto.TaskState,
      Description: dto.Description,
      FromLocation: {
        Building: dto.FromLocation.Building,
        Room: dto.FromLocation.Room,
        X: dto.FromLocation.X,
        Y: dto.FromLocation.Y
      },
      ToLocation: { ...dto.ToLocation },
      ContactInfo: dto.ContactInfo,
      User: dto.User,
      RobotId: dto.RobotId,
      RobotType: dto.RobotType,
      name: dto.name
    };
  }
}

File: jorgecunha07-mgt-0035386b5c0c/jorgecunha07-mgt-0035386b5c0c/Mappers/SurveillanceTaskMapper.cs
using MGT.DTO;
using MGT.Entities;
using MGT.Enums;

namespace MGT.Mappers;

public static class SurveillanceTaskMapper
{
    public static SurveillanceTaskDto ToDto(SurveillanceTask surveillanceTask)
    {
        return new SurveillanceTaskDto
        {
            TaskStatus = surveillanceTask.TaskStatus,
            Description = surveillanceTask.Description ?? string.Empty,
            FromLocation = LocationMapper.ToDto(surveillanceTask.FromLocation),
            ToLocation = LocationMapper.ToDto(surveillanceTask.ToLocation),
            ContactInfo = surveillanceTask.ContactInfo,
            User = surveillanceTask.User,
            RobotId = surveillanceTask.RobotId,
            RobotType = surveillanceTask.RobotType,
            Name = surveillanceTask.Name,
        };
    }

    public static SurveillanceTask ToEntity(SurveillanceTaskCreateDto taskDto)
    {
        return new SurveillanceTask
        {
            TaskStatus = TaskStatusEnum.Submitted,
            Description = taskDto.Description,
            FromLocation = LocationMapper.ToEntity(taskDto.FromLocation),
            ToLocation = LocationMapper.ToEntity(taskDto.ToLocation),
            ContactInfo = taskDto.ContactInfo,
            User = taskDto.User,
            RobotId = taskDto.RobotId,
            RobotType = taskDto.RobotType,
            TaskType = TaskTypeEnum.SurveillanceTask,
            Name = taskDto.Name,
        };
    }
}